apps/mobile/src/common/transactions/stacks-transactions.hooks.ts

```diff
@@ -0,0 +1,51 @@
+import { useCallback } from 'react';
+import { FieldValues } from 'react-hook-form';
+
+import { AccountId } from '@/models/domain.model';
+import { useAccountByIndex } from '@/store/accounts/accounts.read';
+import { useStacksSigners } from '@/store/keychains/stacks/stacks-keychains.read';
+import { useNetworkPreferenceStacksNetwork } from '@/store/settings/settings.read';
+import { bytesToHex } from '@noble/hashes/utils';
+
+import { useNextNonce } from '@leather.io/query';
+import {
+  GenerateUnsignedTransactionOptions,
+  TransactionTypes,
+  generateUnsignedTransaction,
+} from '@leather.io/stacks';
+import { stxToMicroStx } from '@leather.io/utils';
+
+export function useGenerateStxTokenTransferUnsignedTx({ fingerprint, accountIndex }: AccountId) {
+  const network = useNetworkPreferenceStacksNetwork();
+  const account = useAccountByIndex(fingerprint, accountIndex);
+  const stxSigner = useStacksSigners().fromAccountIndex(fingerprint, accountIndex)[0];
+  const stxAddress = stxSigner?.address ?? '';
+  const { data: nextNonce } = useNextNonce(stxAddress);
+
+  return useCallback(
+    async (values?: FieldValues) => {
```

Comment:
This fn is coupled to the form library. Is it accurate that we only ever need to create transactions from a user's input?

Consider increase fee/cancel transaction.